File: sign-of-the-product-of-an-array/sign-of-the-product-of-an-array.cpp
class Solution {
public:
    int arraySign(vector<int>& nums) {
        int c0=-1,c1=0;
        for(int i=0;i<nums.size();i++){
            if(nums[i]==0)
                c0++;
            else if(nums[i]<0)
                c1++;
        }
        if(c0>=0)
            return 0;
        else if (c1%2!=0)
            return -1;
        return 1;
    }
};

File: sign-of-the-product-of-an-array/sign-of-the-product-of-an-array.java
class Solution {
    public int arraySign(int[] nums) {
        int count_zero=0,count_negative=0;
        for(int i=0;i<nums.length;i++){
            if(nums[i]==0){
                count_zero++;
            }
            else if(nums[i]<0){
                count_negative++;
            }           
            if(count_zero>0){
                return 0;
            }
        }
        return (count_negative%2==0?1:-1);
        
    }
}